src/sql/src/plan/func.rs

```diff
@@ -488,18 +583,35 @@ impl ParamList {
                 (
                     ParamType::ListAny,
                     Some(ScalarType::List {
-                        element_type: typ, ..
+                        element_type: el_typ,
+                        custom_oid,
                     }),
                 )
-                | (ParamType::ArrayAny, Some(ScalarType::Array(typ)))
                 | (
                     ParamType::MapAny,
                     Some(ScalarType::Map {
-                        value_type: typ, ..
+                        value_type: el_typ,
+                        custom_oid,
                     }),
-                ) => set_or_check_constrained_type(typ).ok()?,
-                (ParamType::ListElementAny, Some(typ)) | (ParamType::NonVecAny, Some(typ)) => {
-                    set_or_check_constrained_type(typ).ok()?
+                ) => {
+                    // We use typ.as_ref().unwrap() rather than binding the
+                    // interior `ScalarType` to a name because of
+                    // https://github.com/rust-lang/rust/issues/65490
+                    let is_custom = typ.as_ref().unwrap().is_custom_type();
+
+                    // Only check OID of custom types because `None` values for
```

Comment:
I don't fully understand this comment!

Reply:
Ah, very fair! I wanted write this, but found it was unsupported:

```rust
match (param, typ) {
                (
                    ParamType::ListAny,
                    Some(
                        typ
                        @
                        ScalarType::List {
                            element_type: el_typ,
                            custom_oid,
                        },
                    ),
                )
                | (
                    ParamType::MapAny,
                    Some(
                        typ
                        @
                        ScalarType::Map {
                            value_type: el_typ,
                            custom_oid,
                        },
                    ),
                ) => {
                    let is_custom = typ.is_custom_type();
```

but the fact that I am binding the fields to names prevents binding the struct to a name